pysrc/pip_resolve.py

```diff
@@ -123,6 +109,20 @@ def create_package_as_root(package, dir_as_root):
             dir_as_root[DEPENDENCIES][package_as_root[NAME]] = package_tree
     return dir_as_root
 
+def create_dir_as_root(req_file_path, NAME, VERSION, DIR_VERSION, DEPENDENCIES, PACKAGE_FORMAT_VERSION):
+    name, version = None, None
+    if os.path.basename(req_file_path) == 'setup.py':
+        with open(req_file_path, "r") as setup_py_file:
+            name, version = setup_file.parse_name_and_version(setup_py_file.read())
+
+    dir_as_root = {
+        NAME: name or os.path.basename(os.path.dirname(os.path.abspath(req_file_path))) or '_root',
```

Comment:
what do you think about `snyk-python-test` name? this name will be displayed in the UI as a project and `_root` might be unclear

```suggestion
        NAME: name or os.path.basename(os.path.dirname(os.path.abspath(req_file_path))) or 'snyk-python-test',
```

Reply:
_root is default behaviour in dep_graph when there is no rootPkg but this will cause problems if every project is ran from a root directory anyway so discussing